components/dashboard/src/start/VerifyModal.tsx

```diff
@@ -0,0 +1,201 @@
+        const isTokenFilled = () => {
+            return state.token && /\d{6}/.test(state.token);
+        };
+        const verifyToken = async () => {
+            const verified = await getGitpodService().server.verifyPhoneNumberVerificationToken(
+                state.phoneNumber!,
+                state.token!,
+            );
+            if (verified) {
+                setState({
+                    ...state,
+                    verified: true,
+                    message: undefined,
+                });
+            } else {
+                setState({
+                    ...state,
+                    message: {
+                        type: "error",
+                        text: `Invalid verification code.`,
+                    },
+                });
+            }
+            return verified;
+        };
+
+        const reset = () => {
+            setState({
+                ...state,
+                sent: undefined,
+                message: undefined,
+                token: undefined,
+            });
+        };
+        return (
+            <Modal
+                onClose={() => {}}
+                closeable={false}
+                onEnter={verifyToken}
+                title="User Validation Required"
+                buttons={
+                    <div>
+                        <button className="ml-2" disabled={!isTokenFilled()} onClick={verifyToken}>
+                            Validate Account
+                        </button>
+                    </div>
+                }
+                visible={true}
+            >
+                <Alert type="warning" className="mt-2">
+                    To use Gitpod you'll need to validate your account with your phone number. This is required to
+                    discourage and reduce abuse on Gitpod infrastructure.
+                </Alert>
+                <div className="pt-4">
+                    <button className="gp-link" onClick={reset}>
+                        &larr; Use a different phone number
+                    </button>
+                </div>
+                <div className="pt-4">
+                    Enter the verification code we sent to {state.phoneNumber}.<br />
+                    Having trouble?{" "}
+                    <a className="gp-link" href="https://www.gitpod.io/contact/support">
+                        Contact support
+                    </a>
+                </div>
+                {state.message ? (
+                    <Alert type={state.message.type} className="mt-4 py-3">
+                        {state.message.text}
+                    </Alert>
+                ) : (
+                    <></>
+                )}
+                <div className="mt-4">
+                    <h4>Verification Code</h4>
+                    <input
+                        autoFocus={true}
+                        className="w-full"
+                        type="text"
+                        placeholder="enter token send via sms"
```

Comment:
**issue**: Generally, placeholders are problematic for multiple reasons. Some older browsers don't support them[[1](https://caniuse.com/input-placeholder)] although this can be considered minor, some browsers hide them on focus, it's harder for users to check if their input complies with the placeholder hint, have lower contrast[[2](https://usecontra.st/a8a29e/fff)] which is easy to fail for color contrast accessibility, some screen readers don’t recognise them, and much more.

**suggestion**: Instead, we could use a help text below or above the input when possible. In this case, we can safely drop this placeholder text and rely on the input label and the instructions above for users to understand what action they need to take. That's the primary reason why the design specs in https://github.com/gitpod-io/gitpod/issues/11339#issuecomment-1183387915 do not include any placeholders.

```suggestion
```

Reply:
We only support the current version of chrome, firefox and safari.